HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java

```diff
@@ -151,8 +153,24 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
                 Iterator<PlatformCredential> it = pcs.iterator();
                 while (it.hasNext()) {
                     PlatformCredential pc = it.next();
-                    validations.add(validatePlatformCredentialAttributes(pc, device.getDeviceInfo(),
-                            ec));
+                    SupplyChainValidation attributeScv = validatePlatformCredentialAttributes(
+                            pc, device.getDeviceInfo(), ec);
+
+                    if (platformScv != null) {
+                        if (platformScv.getResult() == AppraisalStatus.Status.FAIL
+                                || platformScv.getResult() == AppraisalStatus.Status.ERROR) {
+                            if (attributeScv.getResult() ==  AppraisalStatus.Status.PASS) {
```

Comment:
Is this `attributeScv` guaranteed to be non-null if `platformScv` isn't?

Reply:
I'm not 100% certain, but I was just going back over this and from all indications validatePlaformCredentialAttributes via buildValidationRecord return the object SupplyChainValidation.  I'm fine either way.

Reply:
Actually, based on how this reads, I don't think this update will work in all instances. Having the single instance of `platformScv` won't work when we have multiple Platform Credentials to validate. The last PC provided by the Iterator in the first block will be the one with that sets `platformScv` object.

Instead, a mapping is needed from `PlatformCredential`s to `SupplyChainValidation`s. Then you as you iterate through the Platform Credential Attribute Supply Chain Validation loop that you have here, you can get the corresponding `scv` for the current Platform Credential and perform the check as you are.